src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,74 @@
 package core.basesyntax;
 
 public class SalaryInfo {
+    private static final String SEPARATOR_DATA = " ";
+    private static final String REPORT_INFO = "Report for period ";
+    private static final String ERROR_DATA = "Wrong date format";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        if ((names.length == 0) || (data.length == 0) || (dateFrom.length() == 0)
+                || (dateTo.length() == 0)) {
+            return "";
+        }
+        int dateFromFormat = getDataToIntFormat(dateFrom);
+        int dateToFormat = getDataToIntFormat(dateTo);
+        if (dateFromFormat > dateToFormat) {
+            return "";
+        }
+
+        StringBuilder result = new StringBuilder(REPORT_INFO + dateFrom + " - " + dateTo);
+        for (String wordNameAr : names) {
+            int summaName = 0;
+            for (String datum : data) {
+                StringBuilder builder = new StringBuilder(datum);
+                if (builder.length() == 0) {
+                    continue;
+                }
+                int indexSeparator1 = builder.indexOf(SEPARATOR_DATA);
+                if (indexSeparator1 < 0) {
+                    continue;
+                }
+                int indexSeparator2 = builder.indexOf(SEPARATOR_DATA, indexSeparator1 + 1);
+                if (indexSeparator2 < 0) {
+                    continue;
+                }
+                int wordData = getDataToIntFormat(builder.substring(0, indexSeparator1));
+                String wordName = builder.substring(indexSeparator1 + 1, indexSeparator2);
+                if ((wordNameAr.equals(wordName)) && (wordData >= dateFromFormat)
```

Comment:
Use just one 'if' inside loops and no any 'continue'. Think about how to make it shorter. And use LocalDate.